Fix CREATE EXTENSION IF NOT EXISTS and error messages

erimatnor opened the pull request:
Issuing "CREATE EXTENSION IF NOT EXISTS timescaledb;" generates
an error if the extension is already loaded. This change makes
this command work as expected, with a NOTICE.

Error messages related to CREATE EXTENSION have also been
updated to better adhere to the official PostgreSQL style
guide and also returns the same or similar error codes
and messages as regular PostgreSQL.

cevian commented on src/loader/loader.c:
```diff
@@ -91,41 +91,71 @@ drop_statement_drops_extension(DropStmt *stmt)
+	if (!loaded || !is_extension)
+		return true;
+
+	if (stmt->if_not_exists)
```
What happens if you have an extension loaded, drop the extension, and create it again with a different version using `IF NOT EXISTS`. My reading of this code is that it will allow you to do so even though that is wrong. I think we should have this case tested.

cevian commented on test/expected/loader.out:
```diff
@@ -301,7 +301,7 @@ ERROR:  Mismatched timescaledb version. Shared object file mock-4-mismatch, SQL
 --should not allow since the errored-out mock-4 above already poisoned the well.
 CREATE EXTENSION timescaledb VERSION 'mock-5';
-ERROR:  Cannot create the extension after the another version has already been loaded
+ERROR:  extension "timescaledb" already exists
```
Also this error message seems misleading. TimescaleDB does not exist, but the library has been loaded.

cevian commented on test/expected/loader.out:
```diff
@@ -410,7 +410,7 @@ WARNING:  mock post_analyze_hook "mock-1"
 \set ON_ERROR_STOP 0
 CREATE EXTENSION timescaledb VERSION 'mock-2';
-ERROR:  Cannot create the extension after the another version has already been loaded
+ERROR:  extension "timescaledb" already exists
```
Also misleading comment here.

erimatnor commented:
@cevian addressed the issues you raised. Still, I am a bit concerned that one cannot do:

```
CREATE EXTENSION timescaledb;
DROP EXTENSION timescaledb;
CREATE EXTENSION timescaledb;
```
In the same session. This is incompatible with how standard PostgreSQL works and is something you might want to do in tests (e.g., for a backend service using TimescaleDB), etc.

cevian commented:
@erimatnor The only way I really see of solving that is by not throwing an error on CREATE EXTENSION if the .so has been loaded in the loader (but not loading a new .so either). Then the versioned extension can itself check if any already-loaded .so matches the version being installed. This can be done as one of the first things in a .sql install script or possibly in a hook inside the extension itself.  Let's leave for a separate PR though.
